### PR TITLE
Support for component no decoration (aemNoDecoration)

### PR DESCRIPTION
**Pull request by schoudry**
## Description

Server side html generation in AEM supports cq:NoDecoration. https://experienceleague.adobe.com/docs/experience-manager-65/developing/components/decoration-tag.html?lang=en

This pull request is for adding similar support in React SPA libs...

## Related Issue

https://github.com/adobe/aem-react-editable-components/issues/74

## Motivation and Context

Creating fluid flex layouts with component nesting is becoming complicated due to the wrappers (serving no purpose)

## How Has This Been Tested?

Tested on AEM SDK. The custom component should provide  a sling model for adding aemNoDecoration property in the model json response (Core components sling models were not modified to introduce this property)

image

image

image

image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by schoudry** [comment]
@CezCz @niekraaijmakers could you kindly take a look and provided feedback...

**Review by niekraaijmakers — APPROVED** [review]
You have my seal of approval

File: src/components/EditableComponent.tsx [hunk 1]
```diff
@@ -121,12 +121,26 @@ class EditableComponent<P extends MappedComponentProperties, S extends Container
...
-          </div>
-        );
+        const componentProperties: P = this.props.componentProperties;
+        let renderScript;
+
+        if (!componentProperties.isInEditor && componentProperties.aemNoDecoration){
+            renderScript = (
+                <React.Fragment>
+                  <WrappedComponent {...this.state}/>
+                  <div {...this.emptyPlaceholderProps}/>
```
Comment (andreventuravale) [inline src/components/EditableComponent.tsx:131]: @schoudry maybe we don't need this.

File: src/components/Container.tsx [hunk 2]
```diff
@@ -149,11 +149,24 @@ export class Container<P extends ContainerProperties, S extends ContainerState>
...
-                { this.placeholderComponent }
-            </div>
-        );
+        let renderScript;
+
+        if (!this.props.isInEditor && this.props.aemNoDecoration){
+            renderScript = (
+                <React.Fragment>
+                    { this.childComponents }
+                    { this.placeholderComponent }
```
Comment (andreventuravale) [inline src/components/Container.tsx:158]: @schoudry maybe we don't need this.

File: src/components/Container.tsx [hunk 3]
```diff
@@ -149,11 +149,24 @@ export class Container<P extends ContainerProperties, S extends ContainerState>
...
+                </React.Fragment>
+            )
+        } else {
+            renderScript = (
+                <div {...this.containerProps}>
+                    { this.childComponents }
+                    { this.placeholderComponent }
```
Comment (andreventuravale) [inline src/components/Container.tsx:165]: @schoudry maybe we don't need this.

**Pull request merged**